pandas/core/groupby/grouper.py

```diff
@@ -600,7 +600,7 @@ def _make_codes(self) -> None:
             self._group_index = uniques
 
     @cache_readonly
-    def groups(self) -> Dict[Hashable, np.ndarray]:
+    def groups(self) -> PrettyDict[Hashable, np.ndarray]:
```

Comment:
I *think* this should just be `Dict` - while literally correct we type our API with as permissive of a type as possible for arguments, so I think should do the same on return values as well

Reply:
Agreed on arguments, especially for user-facing methods. But does that also apply to a return, where the result is known to be a PrettyDict? In general I try to be as specific with types as possible; the more specific one can be the more issues mypy can check. I also find the specificity helps with readability.

Reply:
The problem arises when users start using the return type in their own type checking, particularly when things are invariant. For all practical purposes (outside of the repr) this returns a Dict, so I think better downstream to label as such

Reply:
Thanks for the explanation, makes sense. To add to your reasons, I think PrettyDict is also not a public part of the public API. With this, it also shouldn't be leaked out via type-hints.